v1/plugin/stream_collector_proxy.go

```diff
@@ -61,16 +67,18 @@ func (p *StreamProxy) StreamMetrics(stream rpc.StreamCollector_StreamMetricsServ
 	// Metrics out of the plugin into snap.
 	outChan := make(chan []Metric)
 
-	err := p.plugin.StreamMetrics(inChan, outChan, errChan)
-	if err != nil {
-		return err
-	}
+	go func() {
+		err := p.plugin.StreamMetrics(inChan, outChan, errChan)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}()
```

Comment:
@jcooklin, could you provide me more details regardings those changes in `func (p *StreamProxy) StreamMetrics()`? It seems that this cause that [streaming rand collector](https://github.com/intelsdi-x/snap-plugin-lib-go/tree/master/examples/streaming) does not work as expected - running task still has Hits = 0.